ml/Host.cc

```diff
@@ -376,11 +387,32 @@ void DetectableHost::detectOnce() {
             MaxTrainingDuration = std::max(MaxTrainingDuration, DimTrainingDuration);
             TotalTrainingDuration += DimTrainingDuration;
 
-            if (IsAnomalous)
+            if (IsAnomalous) {
                 NumAnomalousDimensions += 1;
+                /*count up the number of anomalies for this dimension*/
+                D->setAnomalousBitCount(D->getAnomalousBitCount() + 1);
+            }
+
+            /*regardless the dimension value was anomalous or not, update the value of the percentage of anomalous dimension*/            
+            D->setAnomalyPercentage((D->getAnomalousBitCount() / (abs(Cfg.SaveAnomalyPercentageEvery - AnomalyBitCounterWindow) * static_cast<double>(updateEvery()))) * 100.0);
```

Comment:
`Cfg.SaveAnomalyPercentageEvery - AnomalyBitCounterWindow` this will be zero in the 1st iteration because `AnomalyBitCounterWindow` is initialized to `Cfg.SaveAnomalyPercentageEvery`, right?

Reply:
Fixed.